### PR TITLE
:art: create build tool to run common task from haskell build tool ther npm

### PR DESCRIPTION
**Pull request by haishengwu-okta**
- also travis could leverage the tool to run test suite in parallel
- split code base to `src` and `app` which has most of implementation and main execution respectively
- clean up scripts from `package.json` so that it gets less changes comparing with generated default.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,10 @@ env:
   global:
     - PHANTOMJS_VERSION=2.1.1
     - NODE_VERSION=5.8
+  matrix:
```
Comment (rchild-okta) [inline .travis.yml]: nice

File: TODO.org [hunk 2]
```diff
@@ -6,5 +6,15 @@
    - or start mock server in an non-blocking process prior tests?
    - or isolate none-pure code from pure code as much as possible
 4. [ ] server side session rather than user cookie to store claims.
-5. [ ] start server from haskell rather npm
+5. [X] start server from haskell rather npm
```
Comment (rchild-okta) [inline TODO.org]: I wonder if it would be more interesting to add these as github issues? Up to you.

File: app/main-buildtool.hs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+             <> header "Welcome to Haskell sample project"
+           )
+
+command_ :: BuildToolOptions -> IO ()
+command_ Clean          = clean
+command_ Lint           = lint
+command_ Stylish        = stylish
+command_ (TestUnit opt) = testunit opt
+command_ TestE2E        = teste2e
+command_ (Start opt)    =   start opt
```
Comment (rchild-okta) [inline app/main-buildtool.hs]: nit: spacing?

File: TODO.org (hunk 2 above)
Comment (haishengwu-okta) [inline TODO.org]: Will do and clean up all todos.

**Pull request merged**